enhancements/kubernetes-nmstate-integration.md

```diff
@@ -0,0 +1,165 @@
+---
+title: kubernetes-nmstate
+authors:
+  - "@schseba"
+reviewers:
+
+approvers:
+  - TBD
+  
+creation-date: 2019-12-18
+last-updated: 2019-12-18
+status: 
+---
+
+# kubernetes-nmstate
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+A proposal to deploy [kubernetes-nmstate](https://github.com/nmstate/kubernetes-nmstate/) on OpenShift.
+
+Node-networking configuration driven by Kubernetes and executed by
+[nmstate](https://nmstate.github.io/).
+
+## Motivation
+
+With hybrid clouds, node-networking setup is becoming even more challenging.
+Different payloads have different networking requirements, and not everything
+can be satisfied as overlays on top of the main interface of the node (e.g.
+SR-IOV, L2, other L2).
+The [Container Network Interface](https://github.com/containernetworking/cni)
+(CNI) standard enables different
+solutions for connecting networks on the node with pods. Some of them are
+[part of the standard](https://github.com/containernetworking/plugins), and there are
+others that extend support for [Open vSwitch bridges](https://github.com/kubevirt/ovs-cni),
+[SR-IOV](https://github.com/hustcat/sriov-cni), and more...
+
+However, in all of these cases, the node must have the networks setup before the
+pod is scheduled. Setting up the networks in a dynamic and heterogenous cluster,
+with dynamic networking requirements, is a challenge by itself - and this is
+what this project is addressing.
+
+### Goals
+
+- Deploy kubernetes-nmstate as part of openshift
+
+### Non-Goals
+
+- Replace SRIOV operator
+
+## Proposal
+
+A new kubernetes-nmstate operator Deployment is deployed to the cluster as part of the the OpenShift
```

Comment:
as a new network operator or as part of the existing network operator?  why is this required on every openshift cluster versus just opt-in to some clusters where the advanced capability is required?  this document should explain why 100% of OpenShift clusters require this capability

Reply:
I think the core problem here is that like a number of other things, nmstate is mostly irrelevant in cloud environments, but really useful for bare metal.  Another exampe is the extension system where `usbguard` is silly in public cloud, but highly relevant in a world in which [plugging in a malicious USB stick to a physical Linux machine can gain arbitrary code execution](https://github.com/coreos/fedora-coreos-tracker/issues/326#issuecomment-565147463).

Reply:
agree that this doesn't seem to be required on every cluster.

If we had a way to say "install this by default on bare metal", I'd recommend adding it to that list.  An operator that's a no-op on cloud clusters doesn't seem ideal, either.  It seems like a OLM managed operator is the only choice we have, unless in some future we can have a bare metal CVO profile or something like that.